app/src/main/java/com/keylesspalace/tusky/db/AccountManager.kt

```diff
@@ -40,11 +41,16 @@ class AccountManager @Inject constructor(db: AppDatabase) {
     private val accountDao: AccountDao = db.accountDao()
 
     init {
-        accounts = accountDao.loadAll().toMutableList()
-
-        activeAccount = accounts.find { acc ->
-            acc.isActive
-        }
+        val disposable = accountDao.loadAll()
```

Comment:
I'm afraid this could cause race conditions? So that `SplashActivity` is checking for existing accounts, but they haven't loaded yet and user would falsely be presented with the login screen?

Reply:
Hm yeah we should make it wait for this probably.

Reply:
It can happen everywhere, when Android recreates an Activity without going over the SplashScreen, e.g. when opening the app from recents. Maybe we should not make the initial load in account manager async.

Reply:
I think the app start is the most painful place regarding blocking right now but I agree that we need to change other places then to expect it async. Should I do it or should we split it into a separate PR?

Reply:
I would leave the initial loading of the account on the main thread
- yes it blocks the thread, but it cannot do anything useful in the meantime anyway - splash screen will be shown even when another thread does the loading
- moving the loading off the main screen means making getAccount somehow async and that will be a lot of changes, it won't be worth it